cpp/unreal_plugins/SimulationController/Source/SimulationController/VehicleAgent.cpp

```diff
@@ -0,0 +1,266 @@
+            Config::get<float>("SIMULATION_CONTROLLER.VEHICLE_AGENT.SPAWN_POSITION_Z"));
+        spawn_rotation = FRotator(
+            Config::get<float>("SIMULATION_CONTROLLER.VEHICLE_AGENT.SPAWN_PITCH"),
+            Config::get<float>("SIMULATION_CONTROLLER.VEHICLE_AGENT.SPAWN_YAW"),
+            Config::get<float>("SIMULATION_CONTROLLER.VEHICLE_AGENT.SPAWN_ROLL"));
+    }
+    else {
+        SP_ASSERT(false);
+    }
+    FActorSpawnParameters actor_spawn_params;
+    actor_spawn_params.Name = Unreal::toFName(Config::get<std::string>("SIMULATION_CONTROLLER.VEHICLE_AGENT.VEHICLE_ACTOR_NAME"));
+    actor_spawn_params.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+    vehicle_pawn_ = world->SpawnActor<AVehiclePawn>(spawn_location, spawn_rotation, actor_spawn_params);
+    SP_ASSERT(vehicle_pawn_);
+
+    auto observation_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.VEHICLE_AGENT.OBSERVATION_COMPONENTS");
+
+    if (Std::contains(observation_components, "camera")) {
+        camera_sensor_ = std::make_unique<CameraSensor>(
+            vehicle_pawn_->camera_component_,
+            Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.VEHICLE_AGENT.CAMERA.RENDER_PASSES"),
+            Config::get<unsigned int>("SIMULATION_CONTROLLER.VEHICLE_AGENT.CAMERA.IMAGE_WIDTH"),
+            Config::get<unsigned int>("SIMULATION_CONTROLLER.VEHICLE_AGENT.CAMERA.IMAGE_HEIGHT"),
+            Config::get<float>("SIMULATION_CONTROLLER.VEHICLE_AGENT.CAMERA.FOV"));
+        SP_ASSERT(camera_sensor_);
+    }
+
+    if (Std::contains(observation_components, "imu")) {
+        imu_sensor_ = std::make_unique<ImuSensor>(vehicle_pawn_->imu_component_);
+        SP_ASSERT(imu_sensor_);
+    }
+}
+
+VehicleAgent::~VehicleAgent()
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    auto observation_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.VEHICLE_AGENT.OBSERVATION_COMPONENTS");
+
+    if (Std::contains(observation_components, "imu")) {
+        SP_ASSERT(imu_sensor_);
+        imu_sensor_ = nullptr;
+    }
+
+    if (Std::contains(observation_components, "camera")) {
+        SP_ASSERT(camera_sensor_);
+        camera_sensor_ = nullptr;
+    }
+
+    SP_ASSERT(vehicle_pawn_);
+    vehicle_pawn_->Destroy();
+    vehicle_pawn_ = nullptr;
+}
+
+void VehicleAgent::findObjectReferences(UWorld* world) {}
+
+void VehicleAgent::cleanUpObjectReferences() {}
+
+std::map<std::string, ArrayDesc> VehicleAgent::getActionSpace() const
+{
+    std::map<std::string, ArrayDesc> action_space;
+    auto action_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.VEHICLE_AGENT.ACTION_COMPONENTS");
+
+    if (Std::contains(action_components, "set_drive_torques")) {
+        ArrayDesc array_desc;
+        array_desc.low_ = std::numeric_limits<float>::lowest();
+        array_desc.high_ = std::numeric_limits<float>::max();
+        array_desc.shape_ = { 4 };
```

Comment:
`{4}` for consistency with other `Agents`

Reply:
Updated

Reply:
Not updated in the code I'm looking at.